src/modm/driver/inertial/bmi088_impl.hpp

```diff
@@ -0,0 +1,355 @@
+/*
+ * Copyright (c) 2023, Christopher Durand
+ *
+ * This file is part of the modm project.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+// ----------------------------------------------------------------------------
+
+#include <cstdint>
+#ifndef MODM_BMI088_HPP
+#error "Don't include this file directly, use 'bmi088.hpp' instead!"
+#endif
+
+#include <modm/math/utils.hpp>
+
+namespace modm
+{
+
+template<Bmi088Transport Transport>
+template<typename... Args>
+Bmi088<Transport>::Bmi088(Args... transportArgs)
+	: Transport{transportArgs...}
+{
+}
+
+template<Bmi088Transport Transport>
+bool
+Bmi088<Transport>::initialize(bool runSelfTest)
+{
+	Transport::initialize();
+
+	if (!checkChipId() or !reset() or !enableAccelerometer()) {
+		return false;
+	}
+
+	if (runSelfTest and !selfTest()) {
+		return false;
+	}
+
+	timerWait();
+	const bool ok = this->writeRegister(GyroRegister::InterruptControl,
+										uint8_t(GyroInterruptControl::DataReady));
+	timer_.restart(WriteTimeout);
+	return ok;
+}
+
+template<Bmi088Transport Transport>
+std::optional<bmi088::AccData>
+Bmi088<Transport>::readAccData()
+{
+	const auto data = this->readRegisters(AccRegister::DataXLow, 6);
+
+	if (data.empty()) {
+		return {};
+	}
+
+	return AccData {
+		.raw = Vector3i(
+			data[0] | data[1] << 8,
+			data[2] | data[3] << 8,
+			data[4] | data[5] << 8
+		),
+		.range = accRange_
+	};
+}
+
+template<Bmi088Transport Transport>
+bool
+Bmi088<Transport>::readAccDataReady()
+{
+	const auto value = readRegister(AccRegister::Status).value_or(0);
```

Comment:
```suggestion
	const auto value = this->readRegister(AccRegister::Status).value_or(0);
```
or why is there a mix of readRegister with `this->` and without?

Reply:
The reason is a C++ quirk. Names from a base class which depends on a template parameter require the `this->` to be visible.
For some technical reasons C++ templates are compiled in two passes. Type substitution happens in the second. The `this->` creates a dependent name for which lookup is deferred. I wouldn't have written the `this` otherwise but one could argue to add it everywhere for consistency. I don't have a strong preference on this.

Reply:
No, leave it as is, I vaguely remember this quirk now, we also do that for I2C transactions. Ugh.